packages/block-editor/src/components/block-list/block-popover.js

```diff
@@ -167,7 +167,9 @@ function BlockPopover( {
 		: 'top right left';
 	const stickyBoundaryElement = showEmptyBlockSideInserter
 		? undefined
-		: getScrollContainer( node ) || ownerDocument.body;
+		: getScrollContainer( ownerDocument.defaultView.frameElement ) ||
+		  getScrollContainer( node ) ||
+		  ownerDocument.body;
```

Comment:
Do you think this could use a comment to explain the different levels of fallback?